From: Jean-Yves Gratius <jyg@gumo.fr>
Date: Wed, 16 Dec 2020 09:54:03 +0100
Subject: bugfix when undoing object move while zooming x 2

with zoom = 2, the undo behaviour was inconsistent after moving an object with mouse / keyboard arrows

Applied-Upstream: 2381852cfe037d4decebfa41774f9e3035a122b7
---
 src/g_editor.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/g_editor.c b/src/g_editor.c
index 340f391..7ad0a14 100644
--- a/src/g_editor.c
+++ b/src/g_editor.c
@@ -842,8 +842,8 @@ void *canvas_undo_set_move(t_canvas *x, int selected)
             {
                 gobj_getrect(y, x, &x1, &y1, &x2, &y2);
                 buf->u_vec[i].e_index = indx;
-                buf->u_vec[i].e_xpix = x1;
-                buf->u_vec[i].e_ypix = y1;
+                buf->u_vec[i].e_xpix = x1/x->gl_zoom;
+                buf->u_vec[i].e_ypix = y1/x->gl_zoom;
                 i++;
             }
     }
@@ -853,8 +853,8 @@ void *canvas_undo_set_move(t_canvas *x, int selected)
         {
             gobj_getrect(y, x, &x1, &y1, &x2, &y2);
             buf->u_vec[indx].e_index = indx;
-            buf->u_vec[indx].e_xpix = x1;
-            buf->u_vec[indx].e_ypix = y1;
+            buf->u_vec[indx].e_xpix = x1/x->gl_zoom;
+            buf->u_vec[indx].e_ypix = y1/x->gl_zoom;
         }
     }
     EDITOR->canvas_undo_already_set_move = 1;
@@ -882,10 +882,10 @@ int canvas_undo_move(t_canvas *x, void *z, int action)
                 glist_select(x, y);
                 gobj_getrect(y, x, &x1, &y1, &x2, &y2);
                 EDITOR->canvas_undo_already_set_move = 1;
-                canvas_displaceselection(x, newx-x1, newy - y1);
+                canvas_displaceselection(x, newx-x1/x->gl_zoom, newy - y1/x->gl_zoom);
                 EDITOR->canvas_undo_already_set_move = doing;
-                buf->u_vec[i].e_xpix = x1;
-                buf->u_vec[i].e_ypix = y1;
+                buf->u_vec[i].e_xpix = x1/x->gl_zoom;
+                buf->u_vec[i].e_ypix = y1/x->gl_zoom;
                 if (cl == vinlet_class) resortin = 1;
                 else if (cl == voutlet_class) resortout = 1;
             }
